src/components/Forms/CopyTraderForm/CopyTraderForm.js

```diff
@@ -93,11 +94,16 @@ const CopyTraderForm = ({ provider, onClose }) => {
               version: 2,
               exchangeInternalId: selectedExchange.internalId,
             };
-            dispatch(setProvider(payload2));
+            dispatch(setProvider(payload2, !provider.profitSharing));
             dashlyProviderEnabled();
             userPilotProviderEnabled();
             dispatch(showSuccessAlert("copyt.follow.alert.title", "copyt.follow.alert.body"));
-            onClose();
+            if (provider.profitSharing) {
+              onClose();
```

Comment:
Why not move `onClose` out of the condition since it's always called?

Reply:
Done.